File: Data/FlexFieldDatabase/FlexFieldScriptSettings.cs
using System;
using GoNorth.Services.ImplementationStatusCompare;

namespace GoNorth.Data.FlexFieldDatabase
{
    /// <summary>
    /// Flex Field Script Settings
    /// </summary>
    public class FlexFieldScriptSettings : IImplementationComparable, ICloneable
    {
        /// <summary>
        /// true if the value should not be exported to a script, else false
        /// </summary>
        [ValueCompareAttribute]
        public bool DontExportToScript { get; set; }

        /// <summary>
        /// Additional Script Names
        /// </summary>
        [ValueCompareAttribute]
        public string AdditionalScriptNames { get; set; }

        /// <summary>
        /// Clones the flex field script settings
        /// </summary>
        /// <returns>Cloned settings</returns>
        public object Clone()
        {
            return new FlexFieldScriptSettings {
                DontExportToScript = this.DontExportToScript,
                AdditionalScriptNames = this.AdditionalScriptNames
            };
        }
    }
}

File: wwwroot/js/FlexFieldDatabase/objectForm/flexFieldScriptSettings.js
(function(GoNorth) {
    "use strict";
    (function(FlexFieldDatabase) {
        (function(ObjectForm) {

            /// Seperator for the additional name field values
            ObjectForm.FlexFieldScriptSettingsAdditionalScriptNameSeperator = ",";

            /**
             * Class for a flex field script settings
             * 
             * @class
             */
            ObjectForm.FlexFieldScriptSettings = function() {
                this.dontExportToScript = false;
                this.additionalScriptNames = "";
            }

            ObjectForm.FlexFieldScriptSettings.prototype = {
                /**
                 * Serializes the values to an object
                 * 
                 * @returns {object} Object to deserialize
                 */
                serialize: function() {
                    return {
                        dontExportToScript: this.dontExportToScript,
                        additionalScriptNames: this.additionalScriptNames
                    };
                },

                /**
                 * Deserialize the values from a serialized entry
                 * @param {object} serializedValue Serialized entry
                 */
                deserialize: function(serializedValue) {
                    this.dontExportToScript = serializedValue.dontExportToScript;
                    this.additionalScriptNames = serializedValue.additionalScriptNames;
                }
            }

        }(FlexFieldDatabase.ObjectForm = FlexFieldDatabase.ObjectForm || {}));
    }(GoNorth.FlexFieldDatabase = GoNorth.FlexFieldDatabase || {}));
}(window.GoNorth = window.GoNorth || {}));